lib/enrichment/src/vrl_util.rs

```diff
@@ -89,6 +83,84 @@ pub(crate) fn add_index(
     Ok(index)
 }
 
+/// Takes a static boolean argument and return the value it resolves to.
+fn arg_to_bool(arg: &FunctionArgument) -> std::result::Result<bool, Box<dyn DiagnosticError>> {
+    arg.expr()
+        .as_value()
+        .as_ref()
+        .and_then(|value| match value {
+            Value::Boolean(true) => Some(true),
+            Value::Boolean(false) => Some(false),
+            _ => None,
+        })
+        .ok_or_else(|| {
+            Box::new(vrl::function::Error::ExpectedStaticExpression {
+                keyword: "case_sensitive",
+                expr: arg.expr().clone(),
+            }) as _
+        })
+}
+
+/// Takes a function argument (expected to be a static boolean) and returns a Case.
+fn arg_to_case(arg: &FunctionArgument) -> std::result::Result<Case, Box<dyn DiagnosticError>> {
+    if arg_to_bool(arg)? {
+        Ok(Case::Sensitive)
+    } else {
+        Ok(Case::Insensitive)
+    }
+}
+
+#[derive(Debug)]
+pub(crate) struct EnrichmentTableRecord {
+    pub(crate) table: String,
+    pub(crate) index: Option<IndexHandle>,
+    pub(crate) case_sensitive: Case,
+    pub(crate) enrichment_tables: TableSearch,
+}
+
+/// Create the index into the enrichment table based on the arguments passed into the function..
+pub(crate) fn index_from_args(
+    table: String,
+    registry: &mut TableRegistry,
+    args: &[(&'static str, Option<FunctionArgument>)],
+) -> std::result::Result<EnrichmentTableRecord, Box<dyn DiagnosticError>> {
+    let case_sensitive = args
+        .iter()
+        .find(|(name, _)| *name == "case_sensitive")
```

Comment:
Seeing this pattern, I'm wondering: Aren't argument names unique? Can we convert the `VmArgumentList` structure to internally hold a `HashMap<PositionOrName, Value>`?

Reply:
A newtype exposing a nicer interface would be nice though, if we want to keep it as an array

Reply:
I fundamentally despise newtypes as they just obscure what the type actually is!

Reply:
> Maybe we can still hide this inside the ArgumentList implementation.

We do have such in ArgumentList.

https://github.com/vectordotdev/vector/blob/stephen/vrl-functions/lib/vrl/compiler/src/vm/argument_list.rs#L62

The type being iterated over here is not an `ArgumentList`. This function is called at compile time, the args here are essentially the parameters and associated expressions that have yet to be compiled. This function is pulling out the static values that have been passed as the case_sensitive param.

`ArgumentList` is built up at runtime.